packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -11,8 +11,9 @@ import 'dart:ui' show Size, Locale, TextDirection, hashValues;
 import 'package:flutter/foundation.dart';
 import 'package:flutter/services.dart';
 
+import '_network_image_io.dart'
+  if (dart.library.html) '_network_image_web.dart' as _image_provider;
```

Comment:
nit: I would name the import `network_image` or something like that, instead of `_image_provider`.